test/common/buffer/buffer_fuzz_test.cc

```diff
@@ -0,0 +1,221 @@
+    ASSERT_EQ(0, target_buffer.search(ctxt.strings_.back()->data(), size, 0));
+    break;
+  }
+  case test::common::buffer::Action::kPrependBuffer: {
+    const uint32_t source_index = action.prepend_buffer() % BufferCount;
+    if (target_index == source_index) {
+      break;
+    }
+    Buffer::OwnedImpl& source_buffer = buffers[source_index];
+    const std::string source_contents = source_buffer.toString();
+    target_buffer.prepend(source_buffer);
+    ASSERT_EQ(0, target_buffer.search(source_contents.data(), source_contents.size(), 0));
+    break;
+  }
+  case test::common::buffer::Action::kReserveCommit: {
+    const uint32_t previous_length = target_buffer.length();
+    const uint32_t reserve_length = clampSize(action.reserve_commit().reserve_length());
+    constexpr uint32_t reserve_slices = 16;
+    Buffer::RawSlice slices[reserve_slices];
+    uint32_t allocated_slices = target_buffer.reserve(reserve_length, slices, reserve_slices);
+    uint32_t allocated_length = 0;
+    for (uint32_t i = 0; i < allocated_slices; ++i) {
+      ::memset(slices[i].mem_, 'c', slices[i].len_);
+      allocated_length += slices[i].len_;
+    }
+    const uint32_t target_length =
+        std::min(allocated_length, action.reserve_commit().commit_length());
+    uint32_t shrink_length = allocated_length;
+    while (shrink_length > target_length) {
+      ASSERT(allocated_slices > 0);
+      const uint32_t available = slices[allocated_slices - 1].len_;
+      const uint32_t remainder = shrink_length - target_length;
+      if (available > remainder) {
+        slices[allocated_slices - 1].len_ -= remainder;
+        break;
+      }
+      shrink_length -= available;
+      --allocated_slices;
+    }
+    target_buffer.commit(slices, allocated_slices);
+    ASSERT_EQ(previous_length + target_length, target_buffer.length());
+    break;
+  }
+  case test::common::buffer::Action::kCopyOut: {
+    const uint32_t start = action.copy_out().start() % target_buffer.length();
+    uint8_t copy_buffer[2 * 1024 * 1024];
+    const uint32_t length =
+        std::min(static_cast<uint32_t>(target_buffer.length()),
+                 std::min(action.copy_out().length(), static_cast<uint32_t>(sizeof(copy_buffer))));
+    target_buffer.copyOut(start, length, copy_buffer);
+    const std::string contents = target_buffer.toString();
+    ASSERT_EQ(0, ::memcmp(copy_buffer, contents.data(), length));
+    break;
+  }
+  case test::common::buffer::Action::kDrain: {
+    const uint32_t previous_length = target_buffer.length();
+    const uint32_t drain_length =
+        std::min(static_cast<uint32_t>(target_buffer.length()), action.drain());
+    target_buffer.drain(drain_length);
+    ASSERT_EQ(previous_length - drain_length, target_buffer.length());
+    break;
+  }
+  case test::common::buffer::Action::kLinearize: {
+    target_buffer.linearize(
+        std::min(static_cast<uint32_t>(target_buffer.length()), action.linearize()));
+    break;
```

Comment:
If you want to validate the linearization results, one lightweight way would be to use `getRawSlices()`:
```
uint32_t linearize_size = std::min(static_cast<uint32_t>(target_buffer.length()), action.linearize());
target_buffer.linearize(linearize_size);
static constexpr NumSlices = 1;
RawSlice slices[NumSlices];
uint64_t slices_used = target_buffer.getRawSlices(slices, NumSlices);
ASSERT_EQ(1, slices_used);
ASSERT_GE(slices[0].len_, linearize_size);
```